sensor_msgs/test/test_point_cloud_msg_wrapper.cpp

```diff
@@ -260,6 +262,107 @@ TEST(PointCloudMsgWrapperTest, point_with_custom_field) {
   ASSERT_EQ(initialized_wrapper.size(), 2U);
 }
 
+struct PointCustomAB
+{
+  uint16_t aa;
+  uint16_t bb;
+  friend bool operator==(const PointCustomAB & p1, const PointCustomAB & p2) noexcept
+  {
+    return p1.aa == p2.aa && p1.bb == p2.bb;
+  }
+};
+LIDAR_UTILS__DEFINE_FIELD_GENERATOR_FOR_MEMBER(aa);
+LIDAR_UTILS__DEFINE_FIELD_GENERATOR_FOR_MEMBER(bb);
+
+/// @test Check check_that_generated_fields_cover_all_point_members
+TEST(PointCloudMsgWrapperTest, point_with_2_custom_fields) {
+  sensor_msgs::msg::PointCloud2 msg1;
+  using Generators = std::tuple<
+    field_aa_generator,
+    field_bb_generator>;
+  using CustomCloudModifier = PointCloud2Modifier<PointCustomAB, Generators>;
+  EXPECT_NO_THROW(CustomCloudModifier(msg1, "some_frame_id"));
+
+  sensor_msgs::msg::PointCloud2 msg2;
+  CustomCloudModifier cloud_wrapper(msg2, "some_frame_id");
+  EXPECT_EQ(msg2.fields[0].name, "aa");
+  EXPECT_EQ(msg2.fields[1].name, "bb");
+}
+
+
+struct PointXYZInt
+{
+  float x;
+  float y;
+  float z;
+  float intensity;
+  PointXYZInt(float x, float y, float z, float intensity)
+  : x(x), y(y), z(z), intensity(intensity) {}
+  PointXYZInt() = default;
+  friend bool operator==(const PointXYZInt & p1, const PointXYZInt & p2) noexcept
+  {
+    return ::nearly_equal(p1.x, p2.x) &&
+           ::nearly_equal(p1.y, p2.y) &&
+           ::nearly_equal(p1.z, p2.z) &&
+           ::nearly_equal(p1.intensity, p2.intensity);
+  }
+};
+
+PointXYZInt PointGenerator()
+{
+  const float multiplier = 0.005;
+  static size_t i = 0;
+  i++;
+  return {i * multiplier,
+      i * multiplier,
+      i * multiplier,
+      i * multiplier};
+}
+
+/// @test Check check_that_generated_fields_cover_all_point_members
+TEST(PointCloudMsgWrapperTest, push_back_time_test) {
+  using Clock = std::chrono::high_resolution_clock;
+  using TimePoint = Clock::time_point;
+
+  auto millis_dif = [](const TimePoint & t1, const TimePoint & t2) {
+      return std::chrono::duration<float, std::milli>(t2 - t1).count();
+    };
+
+  sensor_msgs::msg::PointCloud2 msg_old;
+  sensor_msgs::msg::PointCloud2 msg_new;
+  using Generators = std::tuple<
+    sensor_msgs::field_x_generator,
+    sensor_msgs::field_y_generator,
+    sensor_msgs::field_z_generator,
+    sensor_msgs::field_intensity_generator>;
+  using CloudXYZModifier = PointCloud2Modifier<PointXYZInt, Generators>;
```

Comment:
nit: you don't need custom generators here as all of the fields that you need are part of the default generators. So you could just use the `PointCloud2Modifier<PointXYZInt>` here.